### PR TITLE
add KL divergence measure between policies before/after the param update

### PR DESCRIPTION
**Pull request by watakandai**
# Change
- Added a KL divergence measure to quantify "how different are the policies before/after the policy gradient update"

This should help us to understand why lower controller without entropy works well and why adding entropy breaks the training

### CODE REVIEW

File: pfrl/explorers/additive_gaussian.py [hunk 1]
```diff
@@ -23,6 +23,14 @@ def __init__(self, scale, low=None, high=None):
         self.low = low
         self.high = high
 
+    def distribution(self, x):
+        base_distribution = torch.distributions.Independent(
+            torch.distributions.Normal(loc=x, scale=self.scale), 1
+        )
+        return torch.distributions.transformed_distribution.TransformedDistribution(
+            base_distribution, [torch.distributions.transforms.TanhTransform(cache_size=1)]
```
Comment (peasant98) [inline pfrl/explorers/additive_gaussian.py:31]: this feels a bit confusing because the standard gaussian doesn't use tanh transform, right?

Comment (watakandai) [reply pfrl/explorers/additive_gaussian.py:31]: you are correct. One thing I observed was that, the resulting kl divergence becomes the shape of [batch_size, action_size] without transformedDistribution. Any idea?

Comment (peasant98) [reply pfrl/explorers/additive_gaussian.py:31]: interesting, I don't know why lol

**Comment by peasant98** [comment]
What's the current status of this?

File: pfrl/agents/goal_conditioned_td3.py [hunk 2]
```diff
@@ -265,6 +267,23 @@ def update_policy_with_goal(self, batch):
         self.policy_optimizer.step()
         self.policy_n_updates += 1
 
+        action_distrib = self.policy(torch.cat([batch_state, batch_goal], -1))
+        target_action_distrib = self.target_policy(torch.cat([batch_state, batch_goal], -1))
```
Comment (peasant98) [inline pfrl/agents/goal_conditioned_td3.py]: didn't you want to compare to the policy from the previous iteration, not the target policy?

Comment (watakandai) [reply pfrl/agents/goal_conditioned_td3.py]: We can do both. The reason why I chose the target policy for comparison is because it's more stable than the the policy that is updated every time, meaning that the KL divergence will be quite large for both deterministic and stochastic policies. Instead, if we evaluate current policy against the policy from a while back, I think the difference between deterministic case and stochastic case would be evident. However, we can try both see how it turns out.

Comment (peasant98) [reply pfrl/agents/goal_conditioned_td3.py]: Yea, that would be great.

Comment (watakandai) [reply pfrl/agents/goal_conditioned_td3.py]: fixed

**Review by peasant98 — APPROVED** [review]
LGTM
